api/project/controllers/project.js

```diff
@@ -48,4 +49,73 @@ module.exports = {
     }));
     return sanitizeEntity(entity, {model: strapi.models.project});
   },
+
+  async giveGoogleResources(ctx) {
+    try {
+      const { slug } = ctx.params;
+
+      const project = await strapi.services.project.findOne({ slug });
+
+      const { title, description, team } = project;
+      //formats email based on the title of the project
+      const formatedEmail = strapi.services['google-manager'].formatEmail(title);
+
+      const group = await strapi.services['google-manager'].createGroup(`${formatedEmail}@devlaunchers.com`, description, title);
+
+      if(!Object.keys(team).length === 0) {
```

Comment:
I think it's easier to read if you check length > 0.

Reply:
When would a project not have team?

Reply:
ok

Reply:
> When would a project not have team?

if they don't pass in any leaders or members then they don't have a team

Reply:
Right now all of the projects on the productions site don't have any teams, so if I try to give them google resources we'll get an error.

Reply:
What is the definition of a team?

Reply:
> What is the definition of a team?

A team has leaders and members. If you don't add any leaders or members, strapi returns an empty team object. That's why I check if the team object has any properties to determine if the team is empty.

Reply:
Should we require the team to at least have a leader?

Reply:
> Should we require the team to at least have a leader?

The thing is we don't have all the users on strapi since the frontend just pushed their release yesterday, so they don't have their accounts in there to add themselves as leaders. I've already tested the logic. it should work creating the project without a leader then updating the project later to add a leader. It will add the leader to the google group and give them owner acl.

Reply:
Sounds good to me.